### PR TITLE
feat: support projections in revert statement

### PR DESCRIPTION
**Pull request by mattsse**
I tried to fix this but I believe Hit some lalrpop issues, so this won't build I'm afraid. 
But not sure what the proper fix for this would be, so I'd need some help

Background

this is a valid revert:

```solidity
        function doRevert2() internal pure {
             revert LibName.ErrorName();
        }
```

if an error is defined inside a library `LibName` 

the solidity grammar states the revert stmt is `"revert" + expr + call-arg-list`
But I think only this is valid (Id)? ("." Id)*

### CODE REVIEW

File: solang-parser/src/solidity.lalrpop [hunk 1]
```diff
@@ -121,6 +121,15 @@ StorageLocation: StorageLocation = {
     <l:@L> "calldata" <r:@R> => StorageLocation::Calldata(Loc::File(file_no, l, r)),
 }
 
+IdentifierProject: Vec<Identifier> = {
+    <i:SolIdentifier?> => vec![],
+    <i:SolIdentifier> <v:("." SolIdentifier)*> => {
```
Comment (seanyoung) [inline solang-parser/src/solidity.lalrpop:126]: This is ambiguous: both productions above can produce a single `SolIdentifier`. To fix this:

```
IdentifierProject: Vec<Identifier> = {
    => vec![],
    <i:SolIdentifier> <v:("." SolIdentifier)*> => {
            let mut i = vec![i];
            i.extend(v);
            i
    }
}
```

**Review by seanyoung — CHANGES_REQUESTED** [review]
I think you have the right idea.

File: solang-parser/src/solidity.lalrpop (hunk 1 above)
Comment (mattsse) [reply solang-parser/src/solidity.lalrpop:126]: great, that worked, updated.

**Comment by seanyoung** [comment]
Solved by https://github.com/hyperledger-labs/solang/pull/806

**Pull request closed without merging**